packit/source_git.py

```diff
@@ -74,9 +74,17 @@ def __init__(
         self.source_git = source_git
         self.dist_git = dist_git
         self.upstream_ref = upstream_ref
-        self.upstream_url = upstream_url or next(
-            self.source_git.remote(upstream_remote or "origin").urls
-        )
+        try:
+            self.upstream_url = upstream_url or next(
+                self.source_git.remote(upstream_remote or "origin").urls
+            )
+        except ValueError as exc:
+            logger.debug(f"Failed when getting upstream URL: {exc}")
+            raise PackitException(
+                "Provided source-git repo does not seem to be a clone of the upstream repository "
+                "because there are no git remotes in the repository. Please clone the upstream "
+                "repository or add the origin remote."
```

Comment:
This message doesn't actually tell what went wrong and the fix suggested is somewhat speculative, and might confuse a user which for example makes a typo when providing an `--upstream-remote`.

The `ValueError` is going to be raised, when there is no `--upstream-url` specified and there is no remote found called as the value of `upstream_remote` or "origin". So we should tell the user exactly this.

```suggestion
                "Unable to tell the URL of the upstream repository because there is no remote "
                f"called \'{upstream_remote or 'origin'}\' in {self.source_git.working_dir}. "
                "Please specify the correct upstream remote using '--upstream-remote' or the "
                "upstream URL, using '--upstream-url'."
```